### DISCUSSION TITLE
how to get static variables with ./analyzeHeadless (python api)

### DISCUSSION THREAD

**Discussion by Ruturaj4** [NationalSecurityAgency/ghidra#5122] [Q&A]
I have a simple code:

```
int main(int argc, char *argv[])
{
  static   char buf[10] = "";


  /*  OK  */
  buf[9] = 'A';


  return 0;
}
```
I am using ghidra python api to get the variables used. Something like:

`function.getStackFrame().getStackVariables()`

But, buffer `buf` doesn't show up (it shows up in the ghidra gui, in the code disassembly section). Is there any way to get these using python api? My guess is that, as it is defined `static` it doesn't show up.

Edit:

I found that these variables are defined in the namespaces. So, in the case, buf is defined in `namespaces->main->buf`. I was able to get the namespace for the function - `currentProgram.getSymbolTable().getNamespace(entrypoint)`, but was not able to find the variables.

**Comment by ReversingWithMe** [comment]
The main issue here is Buf lives in a different section and is a global variable, so you wouldn't expect to find it in local variables.

<img width="242" alt="image" src="https://github.com/user-attachments/assets/8f9309e1-c884-4a38-a0fc-c60159320cad" />

```python
from ghidra.program.model.symbol import RefType

listing = currentProgram.getListing()
function = getFunctionContaining(currentAddress)

# Iterate over all instructions in the function
instructions = listing.getInstructions(function.getBody(), True)

for instr in instructions:
    for i in range(instr.getNumOperands()):
        refs = instr.getOperandReferences(i)
        for ref in refs:
            # We're interested in memory read/write references
            if ref.getReferenceType().isData():
                target = ref.getToAddress()
                # Filter by section (global data is usually not in the stack)
                if not target.isStackAddress():
                    print("{} -> references global data at {}".format(instr.getAddress(), target))
```
Debug symbols
<img width="489" alt="image" src="https://github.com/user-attachments/assets/a7993e33-214e-4c07-8198-3d95e3f2d3e1" />

Stripped
<img width="493" alt="image" src="https://github.com/user-attachments/assets/6b76bcfb-61b4-4e3f-ac54-fc5b97268595" />
